from_dict/_caching.py

```diff
@@ -0,0 +1,27 @@
+import functools
+from typing import Optional
+
+from . import _from_dict
+
+_original_function = _from_dict._get_constructor_type_hints
+_cached_function: Optional['functools._lru_cache_wrapper'] = None
```

Comment:
We must not use internals of packages we don't own. (The `_` in `_lru_cache_wrapper` marks it it an internal)

Reply:
Yes, I was concerned about this a bit too. I figured since it was just in a type hint it would not affect the package if it ever changed.
Since caching will always be enabled, this will not be an issue.